tests/test_0184-concatenate-operation.py

```diff
@@ -9,6 +9,26 @@
 
 import awkward1
 
+def test_concatenate_number():
+    assert awkward1.to_list(awkward1.concatenate([
+        awkward1.Array([[1, 2, 3], [], [4, 5]]), 999], axis=1)) == [
+            [1, 2, 3, 999], [999], [4, 5, 999]]
+    assert awkward1.to_list(awkward1.concatenate([
+        awkward1.Array([[[1.1], [2.2, 3.3]], [[]], [[4.4], [5.5]]]), 999], axis=2)) == [
+            [[1.1, 999.0], [2.2, 3.3, 999.0]], [[999.0]], [[4.4, 999.0], [5.5, 999.0]]]
+    assert str(awkward1.type(awkward1.concatenate([
+        awkward1.Array([[1, 2, 3], [], [4, 5]]),
+        awkward1.Array([[123], [223], [323]])], axis=1))) == "3 * var * int64"
+    assert awkward1.to_list(awkward1.concatenate([
+        awkward1.Array([[1, 2, 3], [], [4, 5]]),
+        awkward1.Array([[123], [223], [323]])], axis=1)) == [
+                [1, 2, 3, 123], [223], [4, 5, 323]]
+
+    one = awkward1.Array([[1, 2, 3], [], [4, 5]])
+    two = awkward1.Array([123, 223, 323])
```

Comment:
It's broadcasting; part of this general facility:

```python
>>> ak.Array([10, 20, 30]) + ak.Array([[1, 2, 3], [], [4, 5]])
<Array [[11, 12, 13], [], [34, 35]] type='3 * var * int64'>
```

It applies across the board for all functions of more than one array, so it would be a little strange if concatenation was an exception. Also, using `broadcast_and_apply` will cause axes above the relevant one to be broadcasted, so it's doubly strange for the relevant axis to not be broadcasted. This is just a consistency thing (and likely convenient, since physicists often have arrays of particle parameters and arrays of event parameters, which are at different levels of depth).